source/shared/cpp/ObjectModel/TextBlock.cpp

```diff
@@ -165,3 +174,144 @@ void TextBlock::SetHorizontalAlignment(const HorizontalAlignment value)
+
+        if(matches[Style].matched)
+        {
+            // match for long/short/compact
+            formatStyle = matches[Format].str()[2];
+        }
+
+        parsedostr << matches.prefix().str();
+
+        if(!isDate && formatStyle)
+        {
+            parsedostr << matches[0].str();
+            text = matches.suffix().str();
+            continue;
+        }
+
+        for (unsigned int idx = 0; idx < indexer.size(); idx++)
+        {
+            if (matches[indexer[idx]].matched)
+            {
+                // get indexes for time attributes to index into conrresponding matches
+                // and covert it to string
+                *addrs[idx] = stoi(matches[indexer[idx]]);
+            }
+        }
+
+        // maches offset sign, 
+        // Z == UTC, 
+        // + == time added from UTC
+        // - == time subtracted from UTC
+        if(matches[TimeZone].matched)
+        {
+            
+            char zone = matches[TimeZone].str()[0];
+            factor = (zone == '+')? -1 : 1;
+        }
+
+        // converts to seconds
+        hours *= 3600;
+        minutes *= 60;
+
+        // time zone offset calculation
+        offset = (hours + minutes) * factor;
+
+
+        // measured from year 1900
+        parsedTm.tm_year -= 1900;
+        parsedTm.tm_mon  -= 1;
+
+        time_t utc;
+        // converts to ticks in UTC
+        utc = mktime(&parsedTm);
+        if (utc == -1)
+        {
+            parsedostr << matches[0];
+        }
+
+        char tzOffsetBuff[6] = { 0 };
+        // gets local time zone offset
+        strftime(tzOffsetBuff, 6, "%z", &parsedTm);
+        std::string localTimeZoneOffsetStr(tzOffsetBuff);
+        int nTzOffset = std::stoi(localTimeZoneOffsetStr);
+        offset += ((nTzOffset / 100) * 3600 + (nTzOffset % 100) * 60);
+        // add offset to utc
+        utc += offset;
+        struct tm result = {0};
+
+        // converts to local time from utc
+        if (!LOCALTIME(&result, &utc))
+        {
+            // localtime_s double counts daylight saving time
+            if (result.tm_isdst == 1)
+                result.tm_hour -= 1;
+
+            if (isDate)
+            {
+                switch(formatStyle)
+                {
+                    case 'S':
+                    parsedostr << std::put_time(&result, "%a, %b %e, %Y");
+                    break;
+                    case 'L':
+                    parsedostr << std::put_time(&result, "%A, %B %e, %Y");
+                    break;
+                    case 'C': default:
+                    parsedostr << std::put_time(&result, "%Ex");
+                    break;
+                }
+            }
+            else
+            {
+                parsedostr << std::put_time(&result, "%I:%M %p");
+            }
+        }
+        text = matches.suffix().str();
+        parsedTm = {0};
+        offset = 0;
+        hours = minutes = 0;
+        isDate = false;
+        formatStyle = 0;
```

Comment:
Would it help to define this variables inside the loop? that way you don't have to do this awkward cleanup afer each iteration.

Reply:
yes, but this way,  we don't keep pushing and popping the variable on stack.

---
In reply to: [150951877](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r150951877) [](ancestors = 150951877)

Reply:
I will leave it to style preference I guess, but this seems like a case of over optimizing something that actually makes the code to be more bug prone.

---
In reply to: [150964853](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r150964853) [](ancestors = 150964853,150951877)

Reply:
I agree with Ricardo here.  Let's define our variables in the appropriate scope.

---
In reply to: [150965532](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r150965532) [](ancestors = 150965532,150964853,150951877)

Reply:
ok

---
In reply to: [151218816](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r151218816) [](ancestors = 151218816,150965532,150964853,150951877)